src/StreamFactory/DiactorosStreamFactory.php

```diff
@@ -24,7 +24,7 @@ public function createStream($body = null)
             } else {
                 $stream = new Stream('php://memory', 'rw');
 
-                if (null !== $body) {
+                if (null !== $body || '' !== $body) {
```

Comment:
we could also do `if (null === $body || '' === $body) {return new Stream('php://memory', 'rw');}` as the rest of the code is irrelevant if the body is empty